tool/tsh/common/proxy_test.go

```diff
@@ -477,7 +481,12 @@ func TestWithRsync(t *testing.T) {
 			cmd.Stdout = os.Stdout
 			cmd.Stderr = os.Stderr
 			err = cmd.Run()
-			require.NoError(t, err)
+			if !assert.NoError(t, err) {
+				var exitErr *exec.ExitError
+				if errors.As(err, &exitErr) {
+					t.Logf("exit code: %d", exitErr.ExitCode())
+				}
+			}
```

Comment:
Would be nice if the exit code was included in the test failure as opposed to showing up later in the logs. What about something like this?
```suggestion
			var msg string, exitErr *exec.ExitError
			if errors.As(err, &exitErr) {
				msg = fmt.Sprintf("exit code: %d", exitErr.ExitCode()
			}
			require.NoError(t, err, msg)
```

Reply:
Good idea thanks